### PR TITLE
improve deptool support dependencies for features, support d2 format

### PR DESCRIPTION
**Pull request by Tim-C1**
This is the disscussion post about this PR which contains some displays, views about improvements: https://github.com/orgs/rcore-os/discussions/19

### CODE REVIEW

**Comment by chyyuu** [comment]
Please add a README.md in  tools/deptool/ DIR to introduce how to build, install and use this tool. @Tim-C1

**Comment by Tim-C1** [comment]
> Please add a README.md in tools/deptool/ DIR to introduce how to build, install and use this tool. @Tim-C1

add README for the usage of the tool

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,6 @@
 *.img
 actual.out
 qemu.log
+tools/deptool/Cargo.lock
+tools/deptool/Cargo.lock
```
Comment (equation314) [inline .gitignore:9]: Duplicated item

**Review by chyyuu — APPROVED** [review]
Good work! Please continue to improve this tool to support colors for modules/crates...

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,6 @@
...
+tools/deptool/target
```
Comment (equation314) [inline .gitignore:10]: Create a separate `.gitignore` file in tools?

File: .gitignore (hunk 1 above)
Comment (Tim-C1) [reply .gitignore:9]: ok, I will fix that

File: .gitignore (hunk 2 above)
Comment (Tim-C1) [reply .gitignore:10]: ok

Comment (chyyuu) [reply .gitignore:10]: Done.

**Pull request merged**